### PR TITLE
fix: check for error type before logging

### PR DESCRIPTION
**Pull request by dkoo**
### All Submissions:

* [x] Have you followed the [Newspack Contributing guideline](https://github.com/Automattic/newspack-plugin/blob/trunk/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) and [VIP Go coding standards](https://vip.wordpress.com/documentation/vip-go/code-review-blockers-warnings-notices/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

Fixes a potential fatal. We log an error if a `$lists_response` variable is a WP error OR if it lacks a `lists` key, but in that latter case (not an error and lacks the key) a fatal will be thrown.

### How to test the changes in this Pull Request:

A code review should be sufficient, but to test you can add a line `$lists_response = [];` before [this line](https://github.com/Automattic/newspack-newsletters/blob/trunk/includes/service-providers/mailchimp/class-newspack-newsletters-mailchimp-cached-data.php#L559), then visit a newsletter post in the editor to trigger the fatal on `release` and confirm that the fatal doesn't happen on this branch.

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [x] Have you successfully ran tests with your changes locally?

### CODE REVIEW

File: includes/service-providers/mailchimp/class-newspack-newsletters-mailchimp-cached-data.php [hunk 1]
```diff
@@ -557,7 +557,7 @@ public static function fetch_lists( $limit = null ) {
 			__( 'Error retrieving Mailchimp lists.', 'newspack_newsletters' )
 		);
 		if ( is_wp_error( $lists_response ) || empty( $lists_response['lists'] ) ) {
-			Newspack_Newsletters_Logger::log( 'Mailchimp cache: Error refreshing cache: ' . ( $lists_response->getMessage() ?? __( 'Error retrieving Mailchimp lists.', 'newspack_newsletters' ) ) );
+			Newspack_Newsletters_Logger::log( 'Mailchimp cache: Error refreshing cache: ' . ( is_wp_error( $lists_response ) ? $lists_response->getMessage() : __( 'Error retrieving Mailchimp lists.', 'newspack_newsletters' ) ) );
```
Comment (leogermani) [inline includes/service-providers/mailchimp/class-newspack-newsletters-mailchimp-cached-data.php]: I don't think this is a `WP_Error` object. WP_Error method is `get_message`, not `getMessage`. I think this is something from the Mailchimp lib

Comment (dkoo) [reply includes/service-providers/mailchimp/class-newspack-newsletters-mailchimp-cached-data.php]: How about this, then? f928095

Comment (dkoo) [reply includes/service-providers/mailchimp/class-newspack-newsletters-mailchimp-cached-data.php]: No wait, this 63a9178

Comment (dkoo) [reply includes/service-providers/mailchimp/class-newspack-newsletters-mailchimp-cached-data.php]: Argh, cb62cfe

**Comment by leogermani** [comment]
@dkoo I pushed a change.

The actual problem was when I added `fetch_lists` to the cron callback. 

`->validate()` will either return a successful response or throw an Exception. You can see that all calls to the fetch methods are inside `try catch` clauses. The `getMessage` (camelCase) method comes from the `Exception` object.

You can test this PR by, for example, changing the path of the "fetch lists" request to something invalid at line 556, and then invoke the cron event. You'll see the error being logged safely. Without these changes - and with your previous changes - you would see a Fatal error

**Comment by dkoo** [comment]
Thanks for untangling that, @leogermani!

**Review by claudiulodro — APPROVED** [review]
Solves it nicely in my testing

**Pull request merged**